hollow/src/main/java/com/netflix/hollow/api/client/StaleHollowReferenceDetector.java

```diff
@@ -222,16 +222,17 @@ private boolean shouldDetach() {
 
         private void detach() {
             HollowAPI api = apiHandle.get();
-            HollowDataAccess dataAccess = api.getDataAccess();
-
-            if(dataAccess instanceof HollowProxyDataAccess)
-                ((HollowProxyDataAccess) dataAccess).disableDataAccess();
-            else if (dataAccess instanceof HollowReadStateEngine)
-                ((HollowReadStateEngine) dataAccess).invalidate();
+            if(api != null) {
+                HollowDataAccess dataAccess = api.getDataAccess();
 
-            api.detachCaches();
+                if (dataAccess instanceof HollowProxyDataAccess)
+                    ((HollowProxyDataAccess) dataAccess).disableDataAccess();
+                else if (dataAccess instanceof HollowReadStateEngine)
+                    ((HollowReadStateEngine) dataAccess).invalidate();
 
-            detached = true;
+                api.detachCaches();
+                detached = true;
```

Comment:
should `detach = true` be outside the if condition?

Reply:
ack. If inside, this would lead to redundant calls to shouldDetach().